### PR TITLE
Utils

### PR DESCRIPTION
**Pull request by allanchau**
This PR adds a utils monorepo package.

**Testing**

- [x] Check the GitHub Actions all succeed.

**Deployment**

- [ ] Create a new release.

### CODE REVIEW

File: docusaurus/docs/utils.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+If you need to install a beta version, you can:
+
+```shell
+$ yarn add -E @idearium/utils@beta
+```
+
+## Usage
+
+To use `@idearium/utils`
```
Comment (smebberson) [inline docusaurus/docs/utils.md:26]: @allanchau, this needs to be completed...

File: packages/utils/with-result.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+'use strict';
+
+// This will take a promise and always use `resolve` to return a result in the format `[err, result]`.
```
Comment (smebberson) [inline packages/utils/with-result.js:3]: This could probably be removed from here if you put it into the documentation spot I highlighted above...

**Review by smebberson — CHANGES_REQUESTED** [review]
@allanchau, a couple of things to tidy up.

**Comment by allanchau** [comment]
@smebberson I've tidied this up now.

**Review by smebberson — CHANGES_REQUESTED** [review]
@allanchau, the updates look good. I can't see a workflow for this package though?

**Comment by allanchau** [comment]
@smebberson Oops, I might have accidentally deleted it when I created this branch from the fetch one I am working on. I have fixed it up now

**Comment by smebberson** [comment]
@allanchau, looks good! Please publish.

**Pull request merged**